### ISSUE TITLE
unexpected behavior when using static and const classes

### ISSUE THREAD

**Issue by joelnet** [nikhilk/scriptsharp#42]
In the alert, we are expecting "success", but we are receiving "undefined".

``` csharp
using System;

namespace TestProject
{
    public static class Page
    {
        static Page()
        {
            Class1 c = new Class1();
        }
    }

    public class Class1
    {
        // removing const will make this code succeed.
        const string test = "success";

        public Class1()
        {
            //  expected 'success', but received 'undefined'
            Script.Alert(test);
        }
    }
}
```

This is the output produced, you can see the const value is being set at the bottom of the file.

``` javascript
//! delme.debug.js
//

(function() {

Type.registerNamespace('TestProject');

////////////////////////////////////////////////////////////////////////////////
// TestProject.Page

TestProject.Page = function TestProject_Page() {
}


////////////////////////////////////////////////////////////////////////////////
// TestProject.Class1

TestProject.Class1 = function TestProject_Class1() {
    /// <field name="_test" type="String" static="true">
    /// </field>
    alert(TestProject.Class1._test);
}


TestProject.Page.registerClass('TestProject.Page');
TestProject.Class1.registerClass('TestProject.Class1');
(function () {
    var c = new TestProject.Class1();
})();
TestProject.Class1._test = 'success';
})();

//! This script was generated using Script# v0.7.0.0

```

Also, it seems to be reporting v.0.7.0.0, but the mscorlib.js has this `version: '0.7.1.0'`

**Comment by robitar** [comment]
I run into this a lot, static/const order of initialization is unreliable because static constructors are just executed inline. I can understand why constants would be emitted last, to ensure that the class definitions have already been seen.

I always use something like jQuery.OnDocumentReady(delegate { .. }) inside a static constructor to get around this.

**Comment by nikhilk** [comment]
Fixed in 0.7.4.0 - by virtue of inlining consts.
